pkg/velaql/providers/query/handler.go

```diff
@@ -257,51 +256,30 @@ func (h *provider) GeneratorServiceEndpoints(wfctx wfContext.Context, v *value.V
 				serviceEndpoints = append(serviceEndpoints, generatorFromIngress(ing, cluster)...)
 			}
 		case "SeldonDeployment":
-			var vs istio.VirtualService
-			if err := findResource(&vs, resource.Name, resource.Namespace, resource.Cluster); err != nil {
-				klog.Error(err, fmt.Sprintf("find v1alpha3 VirtualService %s/%s from cluster %s failure", resource.Name, resource.Namespace, resource.Cluster))
-				continue
-			}
-			if vs.Spec.Http == nil || vs.Spec.Http[0].Match == nil {
-				klog.Error(fmt.Sprintf("find VirtualService %s/%s from cluster %s empty prefix", resource.Name, resource.Namespace, resource.Cluster))
-				continue
-			}
-			// get istio service from label here since the seldon's gateway is predefined
-			var istioService corev1.ServiceList
-			labels := &v1.LabelSelector{
-				MatchLabels: map[string]string{
-					"istio": "ingressgateway",
-				},
-			}
-			selector, err := v1.LabelSelectorAsSelector(labels)
-			if err != nil {
-				return err
-			}
-			if err := h.cli.List(ctx, &istioService, &client.ListOptions{
-				LabelSelector: selector,
-			}); err != nil || len(istioService.Items) == 0 {
-				klog.Error(err, fmt.Sprintf("find istio service from cluster %s failure", resource.Cluster))
+			// seldon use ambassador to expose service
+			var service corev1.Service
+			if err := findResource(&service, "ambassador", "vela-system", resource.Cluster); err != nil {
```

Comment:
It's defined here: https://github.com/oam-dev/catalog/pull/257/files#diff-81453dc38f7a27153d8ab982d04d14f69b138dccb8117235915b6437ba95c5d1R25

Reply:
Seldon can use Istio or Ambassador for its ingress, but Seldon can only use Ambassador for custom header routing, so I change the ingress of Seldon to Ambassador. We do not need Istio.VirtualServier here right now since its useless, we can add that when we need.

Reply:
Get the gateway service name and namespace from annotation.